src/Competencies.js

```diff
@@ -0,0 +1,314 @@
+
+    optionalCompetency.push().set({
+      "activated": false,
+      "name": competencyName
+    });
+
+  }
+
+  handleChange(e) {
+    this.setState({ competencyName: e.target.value });
+  }
+
+  handleSave = (e) => {
+    this.setState({
+      visible: false
+    });
+
+    this.addNewCompetency(this.state.competencyName, this.state.option);
+  }
+
+  handleCancel = (e) => {
+
+    this.setState({
+      visible: false,
+    });
+  }
+
+  columnsKMSCore = [{
+    title: 'KMS Core',
+    dataIndex: 'kmscore',
+    key: 'kmscore',
+  }, {
+    title: 'Activated',
+    dataIndex: 'activatedKmscore',
+    key: 'activatedKmscore',
+  }, {
+    title: 'Action',
+    key: 'action',
+    render: (text, record) => (
+      <span>
+        <a onClick={() => this.onSelectCompetency(record.kmscore)}>Edit</a>
+      </span>
+    ),
+  }];
+
+  onSelectCompetency(name) {
+    getData(`competencies1/Kms_core`)
+      .then((dataKmsCore) => _.map(dataKmsCore, (competency, key) => {
+        if (competency.name === name) {
+          getData(`competencies1/Kms_core/${key}`).then((selectedCompetency) => this.setState({
+            optionActivated: selectedCompetency.activated,
+            keyUpdate: key,
+            competencyName: selectedCompetency.name,
+            showEditPopup: true
+          }))
+        }
+      }))
+
+  }
+
+  columnsKMSOptional = [{
+    title: 'KMS Optional',
+    dataIndex: 'kmsoptional',
+    key: 'kmsoptional',
+  }, {
+    title: 'Activated',
+    dataIndex: 'activatedKmsOptional',
+    key: 'activatedKmsOptional',
+  }, {
+    title: 'Action',
+    key: 'action',
+    render: (text, record) => (
+      <span>
+        <a onClick={() => this.onSelectKmsOptionalCompetency(record.kmsoptional)}>Edit</a>
+      </span>
+    ),
+  }];
+
+  onSelectKmsOptionalCompetency(name){
+     getData(`competencies1/Kms_optional`)
+      .then((dataKmsOptional) => _.map(dataKmsOptional, (competency, key) => {
+        if (competency.name === name) {
+          getData(`competencies1/Kms_optional/${key}`).then((selectedCompetency) => this.setState({
+            optionActivatedKmsOptional: selectedCompetency.activated,
+            keyUpdate: key,
+            competencyName: selectedCompetency.name,
+            showEditPopupKmsOptional: true
+          }))
+        }
+      }))
+  }
+
+  getCompetencyKMSCore() {
+    getData(`competencies1/Kms_core`)
+      .then((kmsCoreData) => this.setState({
+        competenciesKMSCore: kmsCoreData
+      }));
+  }
+
+  getCompetencyOptional() {
+    getData(`competencies1/Kms_optional`)
+      .then((kmsOptionalData) => this.setState({
+        competenciesKmsOptional: kmsOptionalData
+      }));
+  }
+
+  componentWillMount() {
+    this.getCompetencyKMSCore();
+    this.getCompetencyOptional();
+
+
+  }
+
+  componentDidMount() {
+
+
+  }
+
+  getDataSoureKMSCore() {
+    _.map(this.state.competenciesKMSCore, (item, index) => {
```

Comment:
const dataSourceCore = _.map(() => return {}) ;
this.setState({dataSourceCode});